packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -286,26 +280,43 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     }
 
     /// @notice Updates the unsafe block signer address. Can only be called by the owner.
+    /// @param _batcherHash New batch hash.
     /// @param _unsafeBlockSigner New unsafe block signer address.
-    function setUnsafeBlockSigner(address _unsafeBlockSigner) external onlyOwner {
+    function setSequencer(bytes32 _batcherHash, address _unsafeBlockSigner) external onlyOwner {
         _setUnsafeBlockSigner(_unsafeBlockSigner);
+        _setBatcherHash(_batcherHash);
+
+        Types.SequencerKeyPair memory _sequencer =
+            Types.SequencerKeyPair({ unsafeBlockSigner: _unsafeBlockSigner, batcherHash: _batcherHash });
+        bytes32 seqHash = Hashing.hashSequencerKeyPair(_sequencer);
+        require(
+            SuperchainConfig(superchainConfig()).allowedSequencers(seqHash),
+            "SystemConfig: Sequencer hash not found in Superchain allow list"
+        );
+    }
+
+    /// @notice Checks the SuperchainConfig's allow list for the current sequencer. If it is not allowed,
+    ///         the sequencer is removed. Anyone may call this function.
+    function checkSequencer() external {
+        Types.SequencerKeyPair memory sequencer =
+            Types.SequencerKeyPair({ unsafeBlockSigner: unsafeBlockSigner(), batcherHash: batcherHash });
+        bytes32 seqHash = Hashing.hashSequencerKeyPair(sequencer);
+        if (SuperchainConfig(superchainConfig()).allowedSequencers(seqHash)) {
+            revert("SystemConfig: cannot remove allowed sequencer.");
```

Comment:
Why a revert here with a string instead of a require?

Reply:
I find this framing to be more readable, ie. saying "if it is allowed, then stop the check", rather than "require that the sequencer is not allowed".
There are other occurences already of the if/revert pattern in our codebase, so IMO readability is sufficient to justify this.